packages/react/src/index.ts

```diff
@@ -141,16 +148,17 @@ export function createActionHook(ctx: Context<Store | null> = context) {
     deps: any[] = [],
   ): (...args: any[]) => void {
     const store = useContext(ctx)
-
     if (!store) throw new Error('[reatom] The provider is not defined')
     if (typeof cb !== 'function') {
       throw new TypeError('[reatom] `useAction` argument must be a function')
     }
 
-    return useCallback((...args) => {
-      const action = cb(...args)
-      if (action) store.dispatch(action)
-    }, deps.concat(store))
+    return batchedFunction(
```

Comment:
Here u neutralizing the useCallback effect and return a new function every time. It will be better to wrap the dispatch to unstable_batchedUpdates